### ISSUE TITLE
Accessibility warnings on controls are misleading

### ISSUE THREAD

**Issue by Josvanegmond** [godotengine/godot#117159]
### Tested versions

- Reproducible since 4.5 (accesskit integration)

### System information

Godot v4.6.stable - Linux Mint 22.3 (Zena) on X11 - X11 display driver, Multi-window, 3 monitors - OpenGL 3 (Compatibility) - NVIDIA GeForce GTX 1050 (nvidia; 535.288.01) - Intel(R) Core(TM) i5-8300H CPU @ 2.30GHz (8 threads) - 23.19 GiB memory

### Issue description

In the editor, there is a setting to turn on accessibility warnings that show up on controls that are missing a text accessibility_name. I think this is a useful feature! If it was just applied to nodes that are relevant. I do want to know if a button is missing it. Having it show up on every container with the text "Accessibility name must not be empty" is incorrect and might have developers implement accessibility incorrectly (like me!).

<img width="278" height="226" alt="Screenshot of godot editor tree view with a dozen control nodes of various types showing a yellow warning triangle" src="https://github.com/user-attachments/assets/73d37909-31f0-4102-a007-72738282da3d" />

You should not be encouraged to add an accessibility name on every container. Most screen readers will probably skip it because it's not an interactive element, but some may not. And some elements are "interactive" like the ScrollContainer where you probably don't want the screen reader to speak its accessibility name.

There are controls that clearly need it, like a button. And there are controls that clearly don't need it, like a vbox. Then there are a few controls that are up for debate: ScrollContainer, and Custom nodes to name two. I think the best choice here would be to display a warning on those, and change the tooltip to "Accessibility name is empty. Consider adding the name and description if this control should be announced by the screen reader."

### Steps to reproduce

N/A

### Minimal reproduction project (MRP)

N/A

**Comment by Josvanegmond** [comment]
> It is indeed not useful for most containers, and even for a lot of interactive controls (like buttons) which have `text`/`title` or similar property that's used as a name.

I did not realize button texts were automatically used by the screen reader and an accessibility name is not needed. It would be nice to have some documentation when and how the accessibility properties should be used. The risk is that we implement it incorrectly and make the screen reader less effective.

Thanks for the swift responses and all the work you've put into the accessibility side of Godot!
